py_mmd_tools/mmd_to_nc.py

```diff
@@ -0,0 +1,256 @@
+        return dict1
+
+    def get_last_metadata_update(self, element):
+        """
+        """
+        out = {
+            'date_metadata_modified': element.find('mmd:update/mmd:datetime',
+                                                   namespaces=self.namespaces).text,
+            'date_metadata_modified_type': element.find('mmd:update/mmd:type',
+                                                        namespaces=self.namespaces).text
+        }
+
+        sep = {
+            'date_metadata_modified': self.mmd_yaml['last_metadata_update']['update']['datetime']
+            ['separator'],
+            'date_metadata_modified_type': self.mmd_yaml['last_metadata_update']['update']['type']
+            ['separator']
+        }
+
+        return out, sep
+
+    def get_personnel(self, element):
+        """
+        """
+
+        out = {}
+        sep = {}
+        for mmd_field in ['role', 'name', 'email', 'organisation']:
+            nc_field = 'contributor_' + mmd_field
+            out[nc_field] = element.find(f'mmd:{mmd_field}', namespaces=self.namespaces).text
+            sep[nc_field] = self.mmd_yaml['personnel'][mmd_field]['separator']
+        return out, sep
+
+    def get_keyword(self, element):
+        """
+        """
+
+        out = {}
+        sep = {}
+        prefix = element.attrib['vocabulary']
+        out['keywords'] = ':'.join([prefix, element.find('mmd:keyword',
+                                                         namespaces=self.namespaces).text])
+        sep['keywords'] = self.mmd_yaml['keywords']['keyword']['separator']
+        try:
+            out['keywords_vocabulary'] = ':'.join([prefix, element.find(
+                'mmd:resource', namespaces=self.namespaces).text])
+            sep['keywords_vocabulary'] = self.mmd_yaml['keywords']['vocabulary']['separator']
+        except AttributeError:
+            print('No resource for keyword')
+
+        return out, sep
+
+    def get_title_abstract(self, element, tag):
+        """
+        """
+
+        out = {}
+        acdd_name, sep = self.get_acdd(self.mmd_yaml[tag][tag])
+        # Check element language
+        for att in element.attrib:
+            if element.attrib[att] == 'no':
+                acdd_name = acdd_name + '_no'
+            elif element.attrib[att] == 'en':
+                out[acdd_name + '_lang'] = 'en'
+            else:
+                print(f'Case not implemented for language {element.attrib[att]}')
+                return None
+        out[acdd_name] = element.text
+        return out
+
+    def get_metadata_identifier(self, element):
+        """
+        """
+
+        out = {}
+        value = element.text
+        try:
+            out['id'] = value.split(':')[1]
+            out['naming_authority'] = value.split(':')[0]
+        except IndexError:
+            out['id'] = value
+
+        return out
+
+    def update_nc(self):
+        """
+        Update a netcdf file global attributes.
+        """
```

Comment:
I'm not sure if I understand what you're doing here. Perhaps we should sit together next week.